src/ToggleAsAClass.jsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+class ToggleAsAClass extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { toggle: props.initialValue };
+
+    // this.switchToggle = this.switchToggle.bind(this);
```

Comment:
If you're using a method defined in a class component remember to either use the `bind` like this or wrap the method in an arrow function! binding or wrapping will make sure that the correct `this`, the class instance, is used in different contexts.